### DISCUSSION TITLE
LSP4J does not compile in Eclipse IDE (com.google.gson cannot be resolved to a module)

### DISCUSSION THREAD

**Discussion by travkin79** [eclipse-lsp4j/lsp4j#855] [General]
Hello everyone,
I hope, someone can give me a hint what I am doing wrong.

I cloned LSP4J (`main` branch, version `0.24.0-SNAPSHOT`), imported and built the Eclipse projects as described in the [contribution guide](https://github.com/eclipse-lsp4j/lsp4j/blob/main/CONTRIBUTING.md). Buildship / gradle successfully built the projects / modules. But the Eclipse IDE / JDT is producing many compile / Java errors, e.g. complaining that "com.google.gson cannot be resolved to a module".

Checking the project and external dependencies in the Package Explorer shows the wired `gson-2.11.0.jar`, so GSON should be available. Maybe JDT does not consider it being a Java module? Does anyone have a clue what I should do to get it working in Eclipse IDE?

Could that be related to issue #702 or to https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2495?

![image](https://github.com/user-attachments/assets/d6804db5-4c3f-4f97-9d87-aeb40e60435b)

Besides, now and then (sporadically) I get the following strange exception that disappears if I repeatedly run the launch config to build the projects.

```
...
> Task :org.eclipse.lsp4j.debug:generateXtext FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':org.eclipse.lsp4j.debug:generateXtext'.
> java.util.zip.ZipException: ZipFile invalid LOC header (bad signature)
```

I'm using
* Eclipse 4.30
* Buildship 3.1.10
* Xtend IDE 2.33.0
* OpenJDK 11.0.18
* RHEL 9.2

**Comment by cdietrich** [comment]
maybe a side effect of the module-java addition.
no idea why this was not properly tested?

**Reply by cdietrich** [reply]
https://github.com/eclipse-lsp4j/lsp4j/pull/843

**Comment by jonahgraham** [comment]
For 0.24.0 I have revered the module-info change. See #868. 

For ongoing discussion about adding module-info, please follow #830
